### ISSUE TITLE
Please add SourceLink data to the nupkg/pdbs

### ISSUE THREAD

**Issue by shueybubbles** [dotnet/SqlClient#2016]
To simplify debugging of `SqlClient` issues in the wild, having source links in the package and PDBs would be very helpful.  See https://github.com/dotnet/sourcelink

**Comment by cheenamalhotra** [comment]
We used to have sourcelinked symbols in SqlClient (you can find them until 4.0.0-preview2 - after that they went missing)
Missing since https://www.nuget.org/packages/Microsoft.Data.SqlClient/4.0.0-preview3.21293.2 🤔

**Comment by paulomorgado** [comment]
Collect a binlog of the build (https://aka.ms/msbuildlog) and see why source linking is not being lost.

**Comment by JRahnama** [comment]
the only related change I can see is PR #1325 between  4.0.0-preview2 and  4.0.0-preview3. I will look into it.

**Comment by cheenamalhotra** [comment]
'BuildForRelease' should be enabled in pipelines, you may wanna review that.

You can also test locally with `/p:BuildForRelease=true` specified it source links are available in PDBs.. but for .NET Core I see it's not source-linking even with enabled flag on latest main (it used to until 4.0.0-p2), so that seems like another issue.

**Comment by paulomorgado** [comment]
Compiling with `dotnet msbuild -bl -restore -t:rebuild -p:Configuration=Release -p:BuildForRelease=true C:\Projects\GH\dotnet\SqlClient\src\Microsoft.Data.SqlClient\netcore\src\Microsoft.Data.SqlClient.csproj` added this to the PDBs:

```JSON
{
  "documents" : {
    "C:\\Projects\\GH\\dotnet\\SqlClient\\*" : "https://raw.githubusercontent.com/dotnet/SqlClient/4657311f9b3f308233bd8682d462351e35bc333e/*"
  }
}
```

The source linking information seems to be added to the PDBs.

I find the whole repo and build overly complex and hard to follow.

**Comment by shueybubbles** [comment]
I concur on the complexity being too much. It's reflected in the nuget package - why do we need separate `ref` assemblies from the `lib` assemblies? Why is there a copy of the netfx binary in both `runtimes` and `lib\net462` folder when they are identical? This build would be easier to maintain with fewer outputs and a simpler package. Developers would like package size reductions for faster builds anyway.

**Comment by paulomorgado** [comment]
@shueybubbles, ref assemblies are good for design-time, because they are smaller and require less design-time resources.

**Comment by David-Engel** [comment]
Done in #2137
